drivers/hmis/app/models/hmis/form/definition_validator.rb

```diff
@@ -146,4 +150,52 @@ def allowed_pick_list_references
       enums.to_set
     end
   end
+
+  def check_cdeds(document, owner_type)
+    cded_check = lambda do |item|
+      (item['item'] || []).each do |child_item|
+        cded_check.call(child_item)
+
+        mapping = child_item['mapping']
+        next unless mapping&.key?('custom_field_key')
+
+        cded_key = mapping['custom_field_key']
+        cded = Hmis::Hud::CustomDataElementDefinition.where(key: cded_key, owner_type: owner_type).first
```

Comment:
performance improvement: do CDED lookups once, instead of in each iteration.
```
cdeds_by_key = Hmis::Hud::CustomDataElementDefinition.where(owner_type: owner_type).index_by(&:key)

or

cded_key_type_hash = Hmis::Hud::CustomDataElementDefinition.where(owner_type: owner_type).pluck(:key, :field_type).to_h
```